### PR TITLE
Enable Solana on mobile via message system feature

### LINKED ISSUE #15613: Prepare feature flag for Solana in message-system
Add possibility to unhide/hide SOL in app via message-system same as we already have via feature flag in Dev menu. 

For that, we have to fix our global handling of feature flags from message system so that only killswitch open full screen blocking modal.

### PR DESCRIPTION
**Pull request by vytick**
- feature messages are not blocking entire app anymore apart of killswitch
- Solana can be enabled/disabled in mobile via message system
- Disabling message to be deployed via #15701

Resolve #15613

### CODE REVIEW

File: suite-native/message-system/src/components/KillswitchMessageScreen.tsx [hunk 1]
```diff
@@ -32,31 +34,24 @@ const buttonsWrapperStyle = prepareNativeStyle(_ => ({
...
     const dispatch = useDispatch();
     const openLink = useOpenLink();
-
-    const activeFeatureMessages = useSelector(selectActiveFeatureMessages);
-    const firstFeatureMessage = A.head(activeFeatureMessages);
-
     const { applyStyle } = useNativeStyles();
 
-    if (!firstFeatureMessage) return null;
+    const killswitch = A.head(useSelector(selectActiveKillswitchMessages));
```
Comment (PeKne) [inline suite-native/message-system/src/components/KillswitchMessageScreen.tsx]: nit: What about changing this selector to `selectFirstActiveKillswitchMessage` and put the `A.head` inside?

File: suite-native/message-system/src/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { createMemoizedSelector, selectActiveFeatureMessages } from '@suite-common/message-system';
+
+export const selectActiveKillswitchMessages = createMemoizedSelector(
```
Comment (matejkriz) [inline suite-native/message-system/src/selectors.ts]: Why don't you return only one killswitch message directly from here? You never need the whole list, always only the highest priority... so `A.head` can be moved here, I guess?

File: suite-native/message-system/src/selectors.ts
Comment (matejkriz) [inline suite-native/message-system/src/selectors.ts:1]: I would prefer not to use such a generic file name, it's hard to orient with many files with the same name. I would prefer `messageSystemSelectors.ts` or `nativeMessageSystemSelectors.ts`.

**Review by PeKne — COMMENTED** [review]
The code looks good  👍

File: suite-native/message-system/src/selectors.ts
Comment (matejkriz) [reply suite-native/message-system/src/selectors.ts:1]: nitpick, no need to fix it until we have a team consensus...

File: suite-native/message-system/src/selectors.ts (hunk 2 above)
Comment (vytick) [reply suite-native/message-system/src/selectors.ts]: 49a065c965

File: suite-native/message-system/src/selectors.ts
Comment (vytick) [reply suite-native/message-system/src/selectors.ts:1]: 49a065c965

File: suite-native/message-system/src/components/KillswitchMessageScreen.tsx (hunk 1 above)
Comment (vytick) [reply suite-native/message-system/src/components/KillswitchMessageScreen.tsx]: 49a065c965

Comment (matejkriz) [reply suite-native/message-system/src/components/KillswitchMessageScreen.tsx]: Even `selectActiveKillswitchMessage` (without the `First`) sounds good to me for this (only one can be active...)

File: suite-native/feature-flags/src/featureFlagsSlice.ts [hunk 3]
```diff
@@ -24,19 +24,19 @@ export const featureFlagsInitialState: FeatureFlagsState = {
...
 };
 
 export const featureFlagsPersistedKeys: Array<keyof FeatureFlagsState> = [
     FeatureFlag.IsDeviceConnectEnabled,
     FeatureFlag.IsRippleSendEnabled,
     FeatureFlag.IsCardanoSendEnabled,
-    FeatureFlag.IsSolanaSendEnabled,
     FeatureFlag.IsRegtestEnabled,
     FeatureFlag.IsSolanaEnabled,
+    FeatureFlag.IsSolanaEnabledByRemote,
```
Comment (matejkriz) [inline suite-native/feature-flags/src/featureFlagsSlice.ts]: Not sure if it makes sense to mix the feature flag from message system with local feature flags in one slice 🤔 It seems like a duplication of the state to me.

File: suite-native/module-accounts-management/src/selectors.ts [hunk 4]
```diff
@@ -1,42 +1,23 @@
...
         state,
         FeatureFlag.IsCardanoSendEnabled,
     );
 
-    if (isCardanoSendEnabled && networkType === 'cardano') return true;
-
-    const isSolanaSendEnabled = selectIsFeatureFlagEnabled(state, FeatureFlag.IsSolanaSendEnabled);
-
-    if (isSolanaSendEnabled && networkType === 'solana') return true;
+    if (networkType !== 'cardano' || isCardanoSendEnabled) return true;
```
Comment (vytick) [inline suite-native/module-accounts-management/src/selectors.ts:20]: send for all the other networkTypes is already enabled, and we will enable sending solana at the same time as we start supporting it. So there is no need to check anything else but cardano FF

**Review by PeKne — APPROVED** [review]

File: suite-native/feature-flags/src/featureFlagsSlice.ts (hunk 3 above)
Comment (vytick) [reply suite-native/feature-flags/src/featureFlagsSlice.ts]: changed

File: suite-native/module-accounts-management/src/selectors.ts (hunk 4 above)
Comment (PeKne) [reply suite-native/module-accounts-management/src/selectors.ts:20]: makes sense to simplify that 👍

File: suite-native/feature-flags/src/featureFlagsSlice.ts [hunk 5]
```diff
@@ -44,6 +41,12 @@ export const featureFlagsSlice = createSlice({
         toggleFeatureFlag: (state, { payload }: PayloadAction<{ featureFlag: FeatureFlag }>) => {
             state[payload.featureFlag] = !state[payload.featureFlag];
         },
+        setFeatureFlag: (
```
Comment (matejkriz) [inline suite-native/feature-flags/src/featureFlagsSlice.ts]: setFeatureFlag is not used anymore

**Review by matejkriz — APPROVED** [review]
Great, it's much simpler now and works correctly, thanks! Just please remove  unused `setFeatureFlag` 🙏

File: suite-native/feature-flags/src/featureFlagsSlice.ts (hunk 5 above)
Comment (vytick) [reply suite-native/feature-flags/src/featureFlagsSlice.ts]: removed

**Comment by vytick** [comment]
Notes for QA @STew790 :  There are two stages of testing, **1.** can be done right away, but lets sync up on testing **2.**

1. App behaviour:
- By default Solana should be disabled
- Solana cant be imported nor is discovered or added new account to device
- enabling local FF should enable it

2. needs to deploy message to develop with enabling Solana and afterwards. App behaviour:
- By default Solana should be enabled
- User should be able to import, discover add and send Solana
- local FF is irrelevant in this case

**Comment by STew790** [comment]
**QA OK**
Tested both behaviors, both worked as expected.
Tested also disabling and works also as expected. :tada:

**Pull request merged**